### PR TITLE
mkvtoolnix 59.0.0

### PR DESCRIPTION
**Pull request by Moisan**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by Moisan** [comment]
From [upstream NEWS.md](https://mkvtoolnix.download/doc/NEWS.md):

> * The Qt library is now required for building all applications, even the
>   command-line ones, as they use Qt's MIME type detection capabilities. In
>   turn this means that you cannot disable the Qt usage anymore; either Qt5 or
>   Qt 6 is required.
> 
>   You can still chose not to build MKVToolNix GUI, though. A new option has
>   been added to `configure` for this purpose: `--disable-gui`.
> * The `gmp` library is now required.
> * The `magic` library is not used anymore.
> * The `PCRE2` & `JPCRE2` libraries are not used anymore. The bundled version
>   of `JPCRE2` was removed.

**Comment by Moisan** [comment]
`qmake` is found but I'm stuck with
```
checking for Qt 6... no: could not compile a test program
```

**Comment by carlocab** [comment]
Look for `CMakeError.log` (it should be in the `logs` artifact) -- that should tell you what it tried to do to test `qt` so you can see where it failed.

**Comment by Moisan** [comment]
The command used includes a `$(EXPORT_ARCH_ARGS)` that shouldn't be there. However I cannot find where that comes from.

**Comment by carlocab** [comment]
I had a look too and couldn't find it. Might be a bug in the `qt` formula.

**Comment by cho-m** [comment]
Maybe try building with `qt@5` to see if issue is related to `qt@6`.

**Comment by Moisan** [comment]
`qt` install has one mention of `EXPORT_ARCH_ARGS` in `share/qt/mkspecs/features/mac/default_post.prf` but I couldn't find a way to change it. `qt@5` seems to fix the issue.

**Comment by carlocab** [comment]
This should probably be reported upstream. Not sure if the right place is `mkvtoolnix` (maybe they're not using `EXPORT_ARCH_ARGS` correctly) or `qt`.

@paperchalice, do you know anything about this?

**Comment by paperchalice** [comment]
There is a related patch https://gitlab.com/mbunkus/mkvtoolnix/-/merge_requests/2234 .

**Comment by Moisan** [comment]
The patch does fix the issue for `qt` but only when running `autogen.sh` first, as we would do when building from HEAD. I tempted to keep `qt@5` for now and upgrade it to `qt` at the next release (with a TODO in the formula).

File: Formula/mkvtoolnix.rb [hunk 1]
```diff
@@ -71,7 +71,7 @@ def install
                           "--with-docbook-xsl-root=#{Formula["docbook-xsl"].opt_prefix}/docbook-xsl",
                           "--with-extra-includes=#{extra_includes}",
                           "--with-extra-libs=#{extra_libs}",
-                          "--disable-qt"
+                          "--disable-gui"
```
Comment (carlocab) [inline Formula/mkvtoolnix.rb:74]: If `qt@5` is still a runtime dep as opposed to build-only, then there's no need to do `--disable-gui`, no?

Comment (Moisan) [reply Formula/mkvtoolnix.rb:74]: My thought was that we wanted to keep tool being command-line only. I'm open to build the GUI if needed.

Comment (carlocab) [reply Formula/mkvtoolnix.rb:74]: Avoiding the GUI when they introduce extra dependencies is usually the right thing to do. Now it seems like the GUI comes for free, so there doesn't seem to be a strong reason to disable it.

Comment (carlocab) [reply Formula/mkvtoolnix.rb]: Oops, that was a lie. We still do need extra dependencies to build the GUI. We can disable it again :)

Comment (Moisan) [reply Formula/mkvtoolnix.rb]: Yep: `configure: error: Could not find the cmark library`

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**